packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts

```diff
@@ -24,11 +38,68 @@ export function inlineEntityOnPluginEvent(event: PluginEvent) {
 
                     // If the entity removed is a readonly entity, try to remove delimiters around it.
                     if (isReadOnly(entity)) {
-                        removeDelimiters(wrapper);
+                        removeDelimiters(entity.wrapper);
                     }
                     break;
             }
             break;
+
+        case PluginEventType.KeyDown:
+        case PluginEventType.KeyUp:
+            if (isCharacterValue(event.rawEvent)) {
+                const position = editor.getFocusedPosition();
```

Comment:
What do we want to do if selection is not collapsed?

getFocusedPosition() will always give you a single position event selection is collapsed.

And what do we want to do for table selection and image selection?

Reply:
Good point, in this case we should only handle it when the selection is collapsed.